nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/resources/docs/org.apache.nifi.processors.script.ExecuteScript/additionalDetails.html

```diff
@@ -0,0 +1,698 @@
+from org.apache.nifi.processor.io import OutputStreamCallback
+
+# Define a subclass of OutputStreamCallback for use in session.write()
+class PyOutputStreamCallback(OutputStreamCallback):
+  def __init__(self):
+        pass
+  def process(self, outputStream):
+    outputStream.write(bytearray('Hello World!'.encode('utf-8')))
+# end class
+flowFile = session.get()
+if(flowFile != None):
+    flowFile = session.write(flowFile, PyOutputStreamCallback())
+# implicit return at the end</pre>
+
+<p><em>Javascript</em></p>
+
+<pre>var OutputStreamCallback =  Java.type("org.apache.nifi.processor.io.OutputStreamCallback");
+var IOUtils = Java.type("org.apache.commons.io.IOUtils");
+var StandardCharsets = Java.type("java.nio.charset.StandardCharsets");
+
+var flowFile = session.get();
+if(flowFile != null) {
+  // Create a new OutputStreamCallback, passing in a function to define the interface method
+  flowFile = session.write(flowFile,
+    new OutputStreamCallback(function(outputStream) {
+        outputStream.write("Hello World!".getBytes(StandardCharsets.UTF_8))
+    }));
+}</pre>
+
+<p><em>JRuby</em></p>
+
+<pre>java_import org.apache.commons.io.IOUtils
+java_import java.nio.charset.StandardCharsets
+java_import org.apache.nifi.processor.io.OutputStreamCallback
+
+# Define a subclass of OutputStreamCallback for use in session.write()
+class JRubyOutputStreamCallback
+  include OutputStreamCallback
+  def process(outputStream)
+    outputStream.write("Hello World!".to_java.getBytes(StandardCharsets::UTF_8))
+  end
+end
+jrubyOutputStreamCallback = JRubyOutputStreamCallback.new
+flowFile = session.get()
+if flowFile != nil
+  flowFile = session.write(flowFile, jrubyOutputStreamCallback)
+end</pre>
+<p>&nbsp;</p>
+<p><strong>Overwrite an incoming flow file with updated content using a callback</strong></p>
+<p><strong>Use Case</strong>: You want to reuse the incoming flow file but want to modify its content for the outgoing flow file.</p>
+<p><strong>Approach</strong>: Use the write(<em>flowFile</em>, <em>streamCallback</em>) method from the session object. An StreamCallback object is needed to pass into the write() method. StreamCallback provides both an InputStream (from the incoming flow file) and an outputStream (for the next version of that flow file), so you can use the InputStream to get the current contents of the flow file, then modify them and write them back out to the flow file. This overwrites the contents of the flow file, so for append you'd have to handle that by appending to the read-in contents, or use a different approach (with session.append() rather than session.write() ). Note that because StreamCallback is an object, the contents are only visible to that object by default. If you need to use the data outside the write() method, use a more globally-scoped variable. The examples will reverse the contents of the incoming flowFile (assumed to be a String) and write out the reversed string to a new version of the flowFile.</p>
```

Comment:
Replace references to "flow file" with "FlowFile".  Also change "flowFile" to "FlowFile"